src/App.js

```diff
@@ -82,8 +84,10 @@ class Machine extends Component {
                 }
             }
         )
+        history.replaceState({source: this.state.source, speed: this.state.speed}, 'Colorfuck', '?s=' + this.state.source + "&speed=" + this.state.speed );
     }
```

Comment:
Same as above. It's probably easiest just to move it into the callback function above.